File: web/src/api/news.js
import service from '@/utils/request'

// ===== 新闻 =====
export const createNews = (data) => {
  return service({
    url: '/promotion/createNews',
    method: 'post',
    data
  })
}

export const updateNews = (data) => {
  return service({
    url: '/promotion/updateNews',
    method: 'put',
    data
  })
}

export const deleteNews = (data) => {
  return service({
    url: '/promotion/deleteNews',
    method: 'delete',
    data
  })
}

export const findNews = (params) => {
  return service({
    url: '/promotion/findNews',
    method: 'get',
    params
  })
}

export const getNewsList = (params) => {
  return service({
    url: '/promotion/getNewsList',
    method: 'get',
    params
  })
}

export const publishNews = (data) => {
  return service({
    url: '/promotion/publishNews',
    method: 'post',
    data
  })
}

export const batchPublishNews = () => {
  return service({
    url: '/promotion/batchPublishNews',
    method: 'post'
  })
}

export const publishNewsCenter = () => {
  return service({
    url: '/promotion/publishNewsCenter',
    method: 'post'
  })
}

export const submitToBaidu = (data) => {
  return service({
    url: '/promotion/submitToBaidu',
    method: 'post',
    data
  })
}

// ===== 新闻分类 =====
export const createNewsCategory = (data) => {
  return service({
    url: '/promotion/createNewsCategory',
    method: 'post',
    data
  })
}

export const updateNewsCategory = (data) => {
  return service({
    url: '/promotion/updateNewsCategory',
    method: 'put',
    data
  })
}

export const deleteNewsCategory = (data) => {
  return service({
    url: '/promotion/deleteNewsCategory',
    method: 'delete',
    data
  })
}

export const findNewsCategory = (params) => {
  return service({
    url: '/promotion/findNewsCategory',
    method: 'get',
    params
  })
}

export const getNewsCategoryList = (params) => {
  return service({
    url: '/promotion/getNewsCategoryList',
    method: 'get',
    params
  })
}

export const getAllEnabledNewsCategories = (params) => {
  return service({
    url: '/promotion/getAllEnabledNewsCategories',
    method: 'get',
    params
  })
}

File: server/router/promotion/news.go
package promotion

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type NewsRouter struct{}

func (s *NewsRouter) InitNewsRouter(Router *gin.RouterGroup) {
	newsRouter := Router.Group("promotion").Use(middleware.OperationRecord())
	{
		newsRouter.POST("createNews", promoApi.NewsApi.CreateNews)
		newsRouter.DELETE("deleteNews", promoApi.NewsApi.DeleteNews)
		newsRouter.PUT("updateNews", promoApi.NewsApi.UpdateNews)
		newsRouter.GET("findNews", promoApi.NewsApi.FindNews)
		newsRouter.GET("getNewsList", promoApi.NewsApi.GetNewsList)
		newsRouter.POST("publishNews", promoApi.NewsApi.PublishNews)
		newsRouter.POST("batchPublishNews", promoApi.NewsApi.BatchPublishNews)
		newsRouter.POST("publishNewsCenter", promoApi.NewsApi.PublishNewsCenter)
		newsRouter.POST("submitToBaidu", promoApi.NewsApi.SubmitToBaidu)
		// 新闻分类
		newsRouter.POST("createNewsCategory", promoApi.NewsCategoryApi.Create)
		newsRouter.DELETE("deleteNewsCategory", promoApi.NewsCategoryApi.Delete)
		newsRouter.PUT("updateNewsCategory", promoApi.NewsCategoryApi.Update)
		newsRouter.GET("findNewsCategory", promoApi.NewsCategoryApi.Find)
		newsRouter.GET("getNewsCategoryList", promoApi.NewsCategoryApi.GetList)
		newsRouter.GET("getAllEnabledNewsCategories", promoApi.NewsCategoryApi.GetAllEnabledCategories)
	}
}

// InitNewsPublicRouter 公开的新闻路由（无需认证）
func (s *NewsRouter) InitNewsPublicRouter(publicRouter *gin.RouterGroup) {
	newsPublic := publicRouter.Group("promotion")
	{
		newsPublic.POST("incrementNewsViewCount", promoApi.NewsApi.IncrementViewCount)
	}
}
